app/tests/src/test_generate.py

```diff
@@ -112,6 +112,34 @@ def test_generate(monkeypatch):
     assert generate("gpt-4o", PROMPT, "some query") == expected_response
 
 
+def test_generate_with_history(monkeypatch):
+    monkeypatch.setattr("src.generate.completion", mock_completion.mock_completion)
+    history = [
+        {"content": "some query", "role": "user"},
+        {"role": "assistant", "content": "<div> answer</div>"},
```

Comment:
Is there a reason the keys are swapped?
```suggestion
        {"content": "<div> answer</div>", "role": "assistant"},
```